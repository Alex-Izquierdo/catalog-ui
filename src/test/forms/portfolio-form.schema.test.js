import { createPortfolioSchema } from '../../forms/portfolio-form.schema';

describe('createPortfolioSchema', () => {
  const workflows = [{
    label: 'foo',
    value: 'bar'
  }, {
    label: 'baz',
    value: 'quxx'
  }];

  it('should create new portfolio form schema variant', () => {
    const schema = createPortfolioSchema(true, workflows);
    expect(schema).toMatchSnapshot();
  });

  it('should create edit portfolio form schema variant', () => {
    const schema = createPortfolioSchema(false, workflows);
    expect(schema).toMatchSnapshot();
  });
});
