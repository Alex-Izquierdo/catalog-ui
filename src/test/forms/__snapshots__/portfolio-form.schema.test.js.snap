// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPortfolioSchema should create edit portfolio form schema variant 1`] = `
Object {
  "fields": Array [
    Object {
      "component": "text-field",
      "isRequired": true,
      "label": "Portfolio Name",
      "name": "name",
      "validate": Array [
        Object {
          "type": "required-validator",
        },
      ],
    },
    Object {
      "component": "textarea-field",
      "label": "Description",
      "name": "description",
    },
    Object {
      "component": "select-field",
      "label": "Approval workflow",
      "name": "workflow_ref",
      "options": Array [
        Object {
          "label": "foo",
          "value": "bar",
        },
        Object {
          "label": "baz",
          "value": "quxx",
        },
      ],
    },
  ],
}
`;

exports[`createPortfolioSchema should create new portfolio form schema variant 1`] = `
Object {
  "fields": Array [
    Object {
      "component": "text-field",
      "isRequired": true,
      "label": "New Portfolio Name",
      "name": "name",
      "validate": Array [
        Object {
          "type": "required-validator",
        },
      ],
    },
    Object {
      "component": "textarea-field",
      "label": "Description",
      "name": "description",
    },
    Object {
      "component": "select-field",
      "label": "Approval workflow",
      "name": "workflow_ref",
      "options": Array [
        Object {
          "label": "foo",
          "value": "bar",
        },
        Object {
          "label": "baz",
          "value": "quxx",
        },
      ],
    },
  ],
}
`;
